llvm/lib/SYCLLowerIR/MutatePrintfAddrspace.cpp

```diff
@@ -125,7 +125,9 @@ Constant *getCASLiteral(GlobalVariable *GenericASLiteral) {
     return ExistingGlobal;
 
   StringRef LiteralValue;
-  getConstantStringInfo(GenericASLiteral, LiteralValue);
+  bool HasString = getConstantStringInfo(GenericASLiteral, LiteralValue);
+  if (!HasString)
+    llvm_unreachable("Unexpected string literal in getCASLiteral()");
```

Comment:
should this be unreachable or assert?

Reply:
I don't have strong opinion on this, but guess in some compilation modes assert() may be translated to NO-OP. Thus I chose here llvm_unreachable(), which seems stronger than assert().
In other place I used assert() - it was rather mimic-ing the code around.